### ISSUE TITLE
SSH option missing from the python client

### ISSUE THREAD

**Issue by agyoungs** [docker/docker-py#3178]
Maybe I missed it, but it appears that the `--ssh` option is missing from the python client. This option provides the SSH agent socket or keys from the host to the build.

Am I missing something obvious here or is this option not provided.

**Comment by naufalafif** [comment]
docker-py is a wrapper for docker engine API & it seems the ssh option is not available. https://docs.docker.com/engine/api/v1.43/#tag/Image/operation/ImageBuild

as workaround, u can use build arg & pass ssh to it.

**Comment by agyoungs** [comment]
@naufalafif Thanks for the reply and I've since realized that the `--ssh` feature was unfortunately not available with the Docker Engine API.

How would I use build args as a workaround? As I understand I would use `ARG SSH` in the Dockerfile, but I'm not sure how I would be able to use the agent in the docker build environment to clone a repo or something similar.

**Comment by milas** [comment]
You're correct - this is not possible using the API via the Python bindings, which only support the classic builder and not BuildKit.

For now, the best option (although not ideal) is to shell out to `docker buildx`.

**Comment by agyoungs** [comment]
> You're correct - this is not possible using the API via the Python bindings, which only support the classic builder and not BuildKit.
> 
> For now, the best option (although not ideal) is to shell out to `docker buildx`.

Is this something that's on the roadmap? I'm happy to contribute if that's helpful

**Comment by kasir-barati** [comment]
Any update @agyoungs @milas @naufalafif.

And @naufalafif  do you mind sharing a snippet for your suggestion [here](https://github.com/docker/docker-py/issues/3178#issuecomment-1740590757), I tried it in Testcontainers and it failed:

```py
import os
from testcontainers.core.image import DockerImage

service_port = 8000
project_root = "/some/path/to/project"
ssh_auth_sock = os.environ.get("SSH_AUTH_SOCK", "")
image = DockerImage(
    path=project_root,
    tag="my-app",
    dockerfile_path="Dockerfile",
    no_cache=True,
    target="prod",
    ssh={"default": [ssh_auth_sock]}, # Pass SSH mount via kwargs to docker-py
)
```

This will fail with this error message:

```
.venv/lib/python3.12/site-packages/testcontainers/core/image.py:96: in __enter__
    return self.build()
           ^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/testcontainers/core/image.py:61: in build
    self._image, self._logs = docker_client.build(
.venv/lib/python3.12/site-packages/testcontainers/core/docker_client.py:55: in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/testcontainers/core/docker_client.py:126: in build
    image_object, image_logs = self.client.images.build(path=path, tag=tag, rm=rm, **kwargs)
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: BuildApiMixin.build() got an unexpected keyword argument 'ssh'
```

> [!NOTE]
>
> In the docstring Testcontainers says it will forward the `kwargs` to `docker-py`: https://github.com/testcontainers/testcontainers-python/blob/b12ae13e589a4ffe326c162a38df56eb30521d69/core/testcontainers/core/image.py#L35

**Comment by agyoungs** [comment]
@kasir-barati No update from me. I don't believe you can do this in a safe way without switching to BuildKit
